### PR TITLE
Add property allowedPrincipalTypes to unifiedRoleDefinition entity

### PR DESCRIPTION
**Pull request by cubika**
Add a new read-only property allowedPrincipalTypes which demonstrates the type of role assignees.

### CODE REVIEW

File: api-reference/beta/resources/unifiedroledefinition.md [hunk 1]
```diff
@@ -36,6 +36,7 @@ The following RBAC providers are currently supported:
 
 | Property     | Type        | Description |
 |:-------------|:------------|:------------|
+|allowedPrincipalTypes|rolePrincipalTypes|Principal types for role assignees, read-only. The possible values are: `user`, `servicePrincipal`, `group`, `unknownFutureValue`. Optional.|
```
Comment (FaithOmbongi) [inline api-reference/beta/resources/unifiedroledefinition.md:39]: - Suggestion
- Given this is a flagged enum, confirm the maximum number of values that can be in the string.
- Also add the property to the JSON representation section

```suggestion
|allowedPrincipalTypes|rolePrincipalTypes|Types of principals that can be assigned the role. Read-only. The possible values are: `user`, `servicePrincipal`, `group`, `unknownFutureValue`. This is a multi-valued enumeration that can contain up to three values as a comma-separated string. For example, `user, group`. Optional.|
```

File: api-reference/v1.0/resources/unifiedroledefinition.md [hunk 2]
```diff
@@ -27,6 +27,7 @@ A role definition is a collection of permissions in Azure Active Directory (Azur
 
 | Property     | Type        | Description |
 |:-------------|:------------|:------------|
+|allowedPrincipalTypes|rolePrincipalTypes|Principal types for role assignees, read-only. The possible values are: `user`, `servicePrincipal`, `group`, `unknownFutureValue`. Optional.|
```
Comment (FaithOmbongi) [inline api-reference/v1.0/resources/unifiedroledefinition.md:30]: Suggestion.
Given this is a flagged enum, confirm the maximum number of values that can be in the string.
Also add to the JSON representation.
```suggestion
|allowedPrincipalTypes|rolePrincipalTypes|Types of principals that can be assigned the role. Read-only. The possible values are: `user`, `servicePrincipal`, `group`, `unknownFutureValue`. This is a multi-valued enumeration that can contain up to three values as a comma-separated string. For example, `user, group`. Optional.|
```

File: api-reference/beta/api/rbacapplication-list-roledefinitions.md [hunk 3]
```diff
@@ -169,6 +169,7 @@ Content-type: application/json
     "value": [
         {
             "id": "729827e3-9c14-49f7-bb1b-9608f156bbb8",
+            "allowedPrincipalTypes": "User, Group",
```
Comment (FaithOmbongi) [inline api-reference/beta/api/rbacapplication-list-roledefinitions.md:172]: GLOBAL: How comes the API returns values in a different casing from the enums casing?

File: changelog/Microsoft.EnterpriseRbac.json [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+          "Description": "Added the `allowedPrincipalTypes` property to the [unifiedRoleDefinition](https://docs.microsoft.com/en-us/graph/api/resources/unifiedRoleDefinition?view=graph-rest-beta) resource.",
+          "Target": "unifiedRoleDefinition"
+        }
+      ],
+      "Id": "6cb9baf7-4b2d-4640-bc60-3f8356c043f3",
+      "Cloud": "Prod",
+      "Version": "beta",
+      "CreatedDateTime": "2022-11-01T07:34:46.8451972Z",
+      "WorkloadArea": "Role management",
+      "SubArea": ""
```
Comment (FaithOmbongi) [inline changelog/Microsoft.EnterpriseRbac.json:27]:
```suggestion
      "WorkloadArea": "Identity and access",
      "SubArea": "Directory management"
```

File: changelog/Microsoft.EnterpriseRbac.json [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+          "Description": "Added the `allowedPrincipalTypes` property to the [unifiedRoleDefinition](https://docs.microsoft.com/en-us/graph/api/resources/unifiedRoleDefinition?view=graph-rest-1.0) resource.",
+          "Target": "unifiedRoleDefinition"
+        }
+      ],
+      "Id": "e32c3f85-25d2-4b20-bf95-25049780a5ef",
+      "Cloud": "Prod",
+      "Version": "v1.0",
+      "CreatedDateTime": "2022-11-01T07:34:46.8452004Z",
+      "WorkloadArea": "Role management",
+      "SubArea": ""
```
Comment (FaithOmbongi) [inline changelog/Microsoft.EnterpriseRbac.json:53]:
```suggestion
      "WorkloadArea": "Identity and access",
      "SubArea": "Directory management"
```

File: changelog/Microsoft.PIM.AzureRBAC.json [hunk 6]
```diff
@@ -1,5 +1,57 @@
...
+          "Description": "Added the `allowedPrincipalTypes` property to the [unifiedRoleDefinition](https://docs.microsoft.com/en-us/graph/api/resources/unifiedRoleDefinition?view=graph-rest-beta) resource.",
+          "Target": "unifiedRoleDefinition"
+        }
+      ],
+      "Id": "8fbabac0-1a40-4318-8c3c-09e2d3fc0861",
+      "Cloud": "Prod",
+      "Version": "beta",
+      "CreatedDateTime": "2022-11-01T07:34:46.8452019Z",
+      "WorkloadArea": "Identity and access",
+      "SubArea": "Identity governance"
```
Comment (FaithOmbongi) [inline changelog/Microsoft.PIM.AzureRBAC.json:27]:
```suggestion
      "WorkloadArea": "Identity and access",
      "SubArea": "Governance"
```

File: changelog/Microsoft.PIM.AzureRBAC.json [hunk 7]
```diff
@@ -1,5 +1,57 @@
...
+          "Description": "Added the `allowedPrincipalTypes` property to the [unifiedRoleDefinition](https://docs.microsoft.com/en-us/graph/api/resources/unifiedRoleDefinition?view=graph-rest-1.0) resource.",
+          "Target": "unifiedRoleDefinition"
+        }
+      ],
+      "Id": "4b984bc5-2c25-4ec8-91a1-9fd91eb9c29a",
+      "Cloud": "Prod",
+      "Version": "v1.0",
+      "CreatedDateTime": "2022-11-01T07:34:46.8452039Z",
+      "WorkloadArea": "Identity and access",
+      "SubArea": "Identity governance"
```
Comment (FaithOmbongi) [inline changelog/Microsoft.PIM.AzureRBAC.json:53]:
```suggestion
      "SubArea": "Governance"
```

**Review by FaithOmbongi — CHANGES_REQUESTED** [review]
Thank you @cubika for your work on this PR.

I've left a few change suggestions inline plus:

- In the unifiedRoleDefinition.md file (beta +v1.0), all the allowedPrincipalTypes property to the JSON representation section.
- Question: How comes the API returns values in a different casing from the enums casing? i.e. "User" instead of "user"?
- Does this new property support $filter. For example with the `has` operator for enums?

File: api-reference/beta/api/rbacapplication-list-roledefinitions.md (hunk 3 above)
Comment (cubika) [reply api-reference/beta/api/rbacapplication-list-roledefinitions.md]: I updated all return values to be camelcase.

**Comment by cubika** [comment]
> Thank you @cubika for your work on this PR.
> 
> I've left a few change suggestions inline plus:
> 
> * In the unifiedRoleDefinition.md file (beta +v1.0), all the allowedPrincipalTypes property to the JSON representation section.
> * Question: How comes the API returns values in a different casing from the enums casing? i.e. "User" instead of "user"?
> * Does this new property support $filter. For example with the `has` operator for enums?

Added the property to the JSON representation. Yes, this property should support $filter, that's the main scenario.

**Comment by FaithOmbongi** [comment]
Hi @cubika - If this PR is ready for a re-review, please select the **re-request review** icon and add the **review feedback addressed** label.

**Pull request closed without merging**